source/common/http/conn_pool_base.cc

```diff
@@ -113,25 +113,32 @@ void MultiplexedActiveClientBase::onGoAway(Http::GoAwayErrorCode) {
 // not considering http/2 connections connected until the SETTINGS frame is
 // received, but that would result in a latency penalty instead.
 void MultiplexedActiveClientBase::onSettings(ReceivedSettings& settings) {
-  if (settings.maxConcurrentStreams().has_value() &&
-      settings.maxConcurrentStreams().value() < concurrent_stream_limit_) {
+  if (settings.maxConcurrentStreams().has_value()) {
     int64_t old_unused_capacity = currentUnusedCapacity();
     // Given config limits old_unused_capacity should never exceed int32_t.
     // TODO(alyssawilk) move remaining_streams_, concurrent_stream_limit_ and
     // currentUnusedCapacity() to be explicit int32_t
     ASSERT(std::numeric_limits<int32_t>::max() >= old_unused_capacity);
-    concurrent_stream_limit_ = settings.maxConcurrentStreams().value();
+    if (configured_stream_limit_ > settings.maxConcurrentStreams().value()) {
```

Comment:
any reason to not set 
concurrent_stream_limit_ = std::min(settings.maxConcurrentStreams().value(), configured_stream_limit_) ?    I think it's a bit easier to read?